File: Python/Math-Array-String/35.py
# 191. Number of 1 Bits

# class Solution:
#     def hammingWeight(self, n: int) -> int:
#         """
#         Returns the number of set bits in the binary representation of n.
#         """
#         count = 0
#         while n:
#             count += n & 1  # Increment count if the least significant bit is set
#             n >>= 1         # Right-shift n by 1 to check the next bit
#         return count

class Solution:
    def hammingWeight(self, n: int) -> int:
        return bin(n).count('1')

sol = Solution()
output = sol.hammingWeight(11)
print(output)

File: JavaScript/Math-Array-String/35.js
// 191. Number of 1 Bits

/**
 * @param {number} n
 * @return {number}
 */
var hammingWeight = function(n) {
    let count = 0;
    while (n !== 0) {
        count += n & 1; // Check if the least significant bit is set
        n = n >>> 1;    // Use unsigned right shift to avoid negative numbers
    }
    return count;
};

// Example usage:
console.log(hammingWeight(11)); // Output: 3 (binary: 1011)
console.log(hammingWeight(128)); // Output: 1 (binary: 10000000)
